deno_core doctests are failing

```
running 4 tests
test ops.rs - ops::json_op_async (line 206) ... ignored
test ops.rs - ops::json_op_sync (line 167) ... ignored
test resources2.rs - resources2::ResourceTable::names (line 129) ... FAILED
test async_cell.rs - async_cell::RcRef (line 110) ... FAILED

failures:

---- resources2.rs - resources2::ResourceTable::names (line 129) stdout ----
error[E0603]: module `resources2` is private
  --> resources2.rs:130:16
   |
4  | use deno_core::resources2::ResourceTable;
   |                ^^^^^^^^^^ private module
   |
note: the module `resources2` is defined here
  --> /Users/biwanczuk/dev/deno/core/lib.rs:19:1
   |
19 | mod resources2;
   | ^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0603`.
Couldn't compile the test.
---- async_cell.rs - async_cell::RcRef (line 110) stdout ----
error[E0603]: module `async_cell` is private
 --> async_cell.rs:112:16
  |
5 | use deno_core::async_cell::RcRef;
  |                ^^^^^^^^^^ private module
  |
note: the module `async_cell` is defined here
 --> /Users/biwanczuk/dev/deno/core/lib.rs:8:1
  |
8 | mod async_cell;
  | ^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0603`.
Couldn't compile the test.

failures:
    async_cell.rs - async_cell::RcRef (line 110)
    resources2.rs - resources2::ResourceTable::names (line 129)

test result: FAILED. 0 passed; 2 failed; 2 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--doc'
```

After removing `ResourceTable` (the old one, we should fix those tests)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -217,7 +217,9 @@ jobs:
 
       - name: Test debug
         if: matrix.kind == 'test_debug'
-        run: cargo test --locked --all-targets
+        run: |
+          cargo test --locked --doc
+          cargo test --locked --all-targets
 
       - name: Run Benchmarks
         if: matrix.kind == 'bench'

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -25,6 +25,8 @@ jobs:
             kind: bench
           - os: ${{ github.repository == 'denoland/deno' && 'ubuntu-latest-xl' || 'ubuntu-18.04' }}
             kind: lint
+          - os: ${{ github.repository == 'denoland/deno' && 'ubuntu-latest-xl' || 'ubuntu-18.04' }}
+            kind: publish-dry-run
 
       # Always run master branch builds to completion. This allows the cache to
       # stay mostly up-to-date in situations where a single job fails due to
@@ -265,6 +267,10 @@ jobs:
             target/release/deno_src.tar.gz
             target/release/lib.deno.d.ts
           draft: true
+
+      - name: Publish Dry Run
+        if: matrix.kind == 'publish-dry-run'
+        run: cd cli && cargo publish --dry-run
 
       - name: Publish
         if: |

--- a/core/async_cell.rs
+++ b/core/async_cell.rs
@@ -109,7 +109,7 @@ impl<T> RcRef<AsyncRefCell<T>> {
 ///
 /// ```rust
 /// # use std::rc::Rc;
-/// # use deno_core::async_cell::RcRef;
+/// # use deno_core::RcRef;
 ///
 /// struct Stuff {
 ///   foo: u32,

--- a/core/resources2.rs
+++ b/core/resources2.rs
@@ -128,8 +128,8 @@ impl ResourceTable {
   /// # Example
   ///
   /// ```
-  /// # use deno_core::resources2::ResourceTable;
-  /// # let resource_table = ResourceTable::default();
+  /// # use deno_core::ResourceTable2;
+  /// # let resource_table = ResourceTable2::default();
   /// let resource_names = resource_table.names().collect::<Vec<_>>();
   /// ```
   pub fn names(&self) -> impl Iterator<Item = (ResourceId, Cow<str>)> {

